accepted/2025/prune-package-reference-rollout.md

```diff
@@ -0,0 +1,69 @@
+
+[`PrunePackageReference`](https://github.com/NuGet/Home/blob/dev/accepted/2024/prune-package-reference.md) allows NuGet to skip platform packages that are already available within the framework that the project is building.
+For example, if your project, targetting .NET 9 has a transitive dependency to System.Text.Json 8.0.4, which is vulnerable, enabling pruning will allow NuGet to skip it. This package will not be downloaded and it will not appear in the dependencies.
+
+We propose that we enable the `PrunePackageReference` feature by default for *all* .NET (Core) and .NET Standard 2.0 and above in .NET 10 preview 1 to reduce false positives by NuGetAudit and related scanning tools and improve performance.
+
+This has 2 benefits:
+
+1. Reduces false positives by NuGet Audit, Component Governance and other dependency auditing tools.
+Our data investigations show that up 30% of transitive vulnerabilities are within packages that may be removed by pruning.
+1. Improves performance, by virtue of having smaller dependency graphs and fewer packages downloaded.
+
+For a feature like this, normally we'd need roll it out based on the target framework, but our telemetry shows that up to 30% of transitive vulnerability warnings may be addressed by enabling the pruning feature.
+The list of packages being removed is the exact same that's part of the build time conflict resolution in the .NET SDK. In a way, this moves the conflict resolution to an earlier part of the build.
+The packages that are being pruned carried assemblies that were excluded from the final output of the project by the aforementioned conflict resolution.
+
+- As we are enabling this feature in .NET 10 Preview 1, it gives a very long runway should we eventually decide to go with a more convervative roll-out strategy.
+- If there are any issues, users can easily disable the feature by setting `RestoreEnablePackagePruning` to `false` in their project/props.
+- We will continuously monitor telemetry and feedback around the feature through all the previews.
+
+### Testing & Validation
+
+The feature has been development since November and the implementation has been extensively tested through automation, but we also ensured that our own repos can dogfood without any issues.
+
+We validated the .NET SDK build that it *does not affect the builds negatively of the following repos*:
+
+- NuGet/NuGet.Client
+- dotnet/sdk
+- dotnet/roslyn - <https://github.com/dotnet/roslyn/pull/76898>
+- dotnet/runtime - <https://github.com/dotnet/roslyn/pull/76896>
+- dotnet/aspnetcore - <https://github.com/dotnet/aspnetcore/pull/60025>
+
+## Risks and drawbacks
+
+It may lead to some build issues for if a build has customization such as:
+
+- If a project manually references an assembly or props/targets from a package from the global packages folder. Note that pruning is not allowed for direct dependencies, so this would need to be a transitive package.
```

Comment:
Someone might be doing this for the purpose of static analysis against older packages.  If they wish to do this they can use PackageDownload.  That will be a more predictable result.  I don't expect anyone to actually do this - we haven't found any instances.

Reply:
That's interesting. I would've expected people to do static analysis on framework assemblies unless I'm misunderstanding your point.

Reply:
For example - API compat.  Suppose I wanted to compare what an old package provided vs new. Definitely not customer facing thing.